### PR TITLE
Develop the new REST methods for 'networking' endpoint

### PR DESCRIPTION
**Pull request by alzinchenko**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/utils/ResponseBuilder.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    protected ResponseBase response;
+    protected int status;
+
+    protected ResponseBuilder(int status) {
+        this.status = status;
+        this.startRequestTime = System.currentTimeMillis();
+    }
+
+    public static ResponseBuilder startTiming(){
+        return new ResponseBuilder(200);
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/utils/ResponseBuilder.java]: Status 200 is correct only for successful requests, for errors required 4xx codes such as 422 for validation

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpointTest.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+class NetworkEndpointTest {
+    private static final String PEER_ADDRESS = "192.168.2.68";
+    private static final String WRONG_PEER_ADDRESS = "10.0.0.1";
+    private static final String ANNOUNCED_ADDRESS = "10.10.10.10";
+
+    private static ObjectMapper mapper = new ObjectMapper();
+    private static Dispatcher dispatcher;
+
+    @BeforeAll
+    static void setupClass(){
```
Comment (AndrewBoyarsky) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpointTest.java]: Setting up mocks for all test methods make it painful to determine, which mock are related for specific test methods. In the 'given' section of each test method you can setup needed mocks .

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/ApiErrors.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ *  Copyright © 2018-2019 Apollo Foundation
+ *
+ */
+
+package com.apollocurrency.aplwallet.apl.core.rest;
+
+public enum ApiErrors implements ErrorInfo {
+
+    JSON_SERIALIZATION_EXCEPTION(1,1001,"Exception encountered during generating JSON content, root cause: %s"),
```
[the remark this answers is not in the document]
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/ApiErrors.java]: This error occurs when building a response namely the JSON generation, it's a new error type. This error doesn't relate to query requirements or parameters validation. That's why I've defined the 1000 family error.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+        }
+
+        boolean isNewlyAdded = service.addPeer(peer, peerAddress);
+        PeerDTO dto = peerConverter.convert(peer);
+        dto.setIsNewlyAdded(isNewlyAdded);
+
+        return response.bind(dto).build();
+    }
+
+    @Path("/peer/all")
```
[the remark this answers is not in the document]
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpoint.java]: sure

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+                            content = @Content(mediaType = "application/json",
+                                    schema = @Schema(implementation = MyPeerInfoDTO.class)))
+            })
+    public Response getMyInfo(@Context HttpServletRequest request) {
+        ResponseBuilder response = ResponseBuilder.startTiming();
+        MyPeerInfoDTO dto = new MyPeerInfoDTO(request.getRemoteHost(), request.getRemoteAddr());
+        return response.bind(dto).build();
+    }
+
+    @Path("/peer")
```
[the remark this answers is not in the document]
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpoint.java]: Yes, it's a basic approach but the task was to develop the new REST 'networking' endpoint in such a way the existing UI works without any changes. At this moment, there is no ability to forward an old-style GET request to the new URL with path param in the ApiSplitFilter class, maybe it needs to be developed.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/utils/ResponseBuilder.java (hunk 1 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/utils/ResponseBuilder.java]: It's reasonable, but as I got, the 'Working UI' is the main requirement. Currently, the UI expects that when the error occurs the errorCode field will set to a non-zero value but the HTTP status will set to 200.

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpointTest.java (hunk 2 above)
Comment (alzinchenko) [reply apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpointTest.java]: Yes, sure

**Pull request merged**